python/pyspark/sql/functions.py

```diff
@@ -152,13 +152,19 @@ def _():
 _collect_list_doc = """
     Aggregate function: returns a list of objects with duplicates.
 
+    .. note:: The function is non-deterministic because the order of collected results depends
+        on order of rows which may be non-deterministic after a shuffle.
```

Comment:
I feel that non-deterministic here is different with other non-deterministic like `monotonically_increasing_id` or `uuid`.

Should we just say `The order of collected results is non-deterministic and depends on order of rows which may be non-deterministic after a shuffle`?

Reply:
Nature of non-determinism can be different but I believe it is important to mention explicitly in the note that the function is `non-deterministic`. It would be common style of notes for all non-deterministic functions. Maybe it is not good example but event for `grep`ing or searching in docs it is pretty convenient.